### ISSUE TITLE
Ruby Heredoc indentation and line width

### ISSUE THREAD

**Issue by iancward** [bbatsov/rubocop#4449]
It looks like Rubocop [now cares about heredoc indentation](https://github.com/bbatsov/rubocop/blob/dcb3f160a2e23218a7826aadd0e64baefd6d7521/lib/rubocop/cop/layout/indent_heredoc.rb); however, sometimes heredocs are left not-indented so as to keep text lines without wrapping. 
Rubocop seems to want to such a heredoc to be indented properly, but then doesn't care that the lines wrap. I understand why someone [would not want rubocop to care about formatting inside the heredoc](https://github.com/bbatsov/rubocop/issues/1407); however, I would think that it should only care about indentation when such indentation would not cause less-than-ideal formating (e.g. line wrapping).

This would follow the spirit of other cops, such as the [GuardClause](https://github.com/bbatsov/rubocop/blob/fdc67866b8e876cfbe9dab0ce342e4fe977606b6/manual/cops_style.md#styleguardclause) cop, which only recommends single-line conditionals when doing so would not exceed 80 characters.

--------

## Expected behavior

Rubocop would not care about proper heredoc indentation when doing so would cause the heredoc to exceed eighty columns in width

## Actual behavior

Rubocop wants me to indent my heredoc (using squiggly style or ActiveSupport) even though doing so would cause lines to wrap. It then does not throw any errors that the lines are wrapped.

## Steps to reproduce the problem

Given a heredoc similar to below, in a block nested four times (in a spec file):
```
<<~EOF
repo id               repo name                                  status
base/x86_64           redhat-7Server - Base                      13,813
chef-stable/x86_64    Chef Software Inc chef-stable repository      292
epel/x86_64           redhat-7Server - Epel                      13,539
extras/x86_64         redhat-7Server - Extras                       140
updates/x86_64        redhat-7Server - Updates                   13,813
EOF
```
Rubocop wants me to indent the EOF to five positions (ten spaces), and the actual heredoc text an extra position. This causes lines to wrap.

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.3.1 x86_64-linux)
```
